app/src/main/java/com/github/se/eduverse/ui/folder/ListFolders.kt

```diff
@@ -119,7 +119,7 @@ fun ListFoldersScreen(
                 try {
```

Comment:
I suggest moving the try catch block to the view model function since it's part of the logic and hence should be done there.
Also you're only setting `selected` to `emptyList()` when the deletion is successful but I think the list should also be emptied when an error occurs